web/modules/custom/joinup_subscription/joinup_subscription.module

```diff
@@ -50,26 +50,49 @@ function joinup_subscription_entity_base_field_info(EntityTypeInterface $entity_
 /**
  * Implements hook_message_digest_aggregate_alter().
  *
- * Reorders the messages in the content subscription digest to be
- * grouped by collection. The collections will be sorted alphabetically and also
- * the community content items within each collection is sorted alphabetically.
+ * Reorders the messages in the group content subscription digest to be grouped
+ * by the parent group. The groups will be sorted alphabetically and also
+ * the community content items within each group is sorted alphabetically.
  */
 function joinup_subscription_message_digest_aggregate_alter(array &$context, UserInterface $account, MessageNotifierInterface $notifier): void {
-  // Only act on the collection community content digest. This can be recognized
+  // Only act on the group community content digest. This can be recognized
   // because it is not linked to any entity type or entity ID.
   if (!empty($context['entity_type']) || !empty($context['entity_id'])) {
     return;
   }
 
+  $fields = [
+    'collection_content_subscription' => 'field_collection_content',
+    'solution_content_subscription' => 'field_solution_content',
+  ];
+
   // Filter out any messages that are referring to deleted content.
   $messages = &$context['messages'];
   foreach ($messages as $key => $message) {
-    $content = $message->field_collection_content->first()->entity ?? NULL;
+    $content = $message->get($fields[$message->getTemplate()->id()])->first()->entity ?? NULL;
     if (empty($content)) {
       unset($messages[$key]);
     }
   }
 
+  if (!empty($message)) {
+    // Message digest does not carry over any information other than the
+    // notifier and even this, is very limited in capabilities.
+    // Set the bundle label as a temporary variable in the notifier as a dynamic
+    // property since it does provide any methods to cary information over.
+    // This will allow us to alter the mail subject and include the group bundle
+    // in it.
+    //
+    // We are accessing information the same way as we do later in the sorting
+    // function so it is safe to assume that values do exist.
+    //
+    // @see \joinup_subscription_mail_alter()
+    $notifier->bundle_label = ucfirst($message->get($fields[$message->getTemplate()->id()])
```

Comment:
Same here, better read the standard label than use `ucfirst()`.

Reply:
Here, I am reading the singular label of the bundle type but it comes in all short. I am just making the first character capital.